File: pinyougou-user-web/src/main/webapp/js/service/userService.js
/**
* @Description 用户service
* @Author  guanx
* @Date   2020/3/1 20:13
* @Param
* @Return
* @Exception
*
*/
app.service('userService',function ($http) {

    //注册功能
    this.reg=function (entity) {
       return $http.post('/user/add.shtml',entity);
    }

    //创建验证码
    this.createCode=function (phone) {
        return $http.get('/user/create/code.shtml?phone=' + phone);
    }
})

File: pinyougou-user-interface/src/main/java/com/pinyougou/user/service/UserService.java
package com.pinyougou.user.service;

import com.pinyougou.model.User;

/**
* @Description 用户注册
* @Author  guanx
* @Date   2020/3/1 18:38
* @Param
* @Return
* @Exception
*
*/
public interface UserService {

    /**
    * @Description 增加用户信息
    * @Author  guanx
    * @Date   2020/3/1 18:53
    * @Param
    * @Return
    * @Exception
    *
    */
    int add(User user);

    /**
    * @Description 创建验证码
    * @Author  guanx
    * @Date   2020/3/2 11:29
    * @Param
    * @Return
    * @Exception
    *
    */
    void createCode(String phone);
}
